motionplan/constraint_handler.go

```diff
@@ -0,0 +1,350 @@
+//go:build !no_cgo
+
+package motionplan
+
+import (
+	"fmt"
+
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+)
+
+var defaultMinStepCount = 2
+
+// ConstraintHandler is a convenient wrapper for constraint handling which is likely to be common among most motion
+// planners. Including a constraint handler as an anonymous struct member allows reuse.
+type ConstraintHandler struct {
+	segmentConstraints   map[string]SegmentConstraint
+	segmentFSConstraints map[string]SegmentFSConstraint
+	stateConstraints     map[string]StateConstraint
+	stateFSConstraints   map[string]StateFSConstraint
```

Comment:
Could consider making these interfaces to cut down on the need for duplication.  But if the vision is we're moving away from the original versions then probably not necessary

Reply:
I think they would need to be generics, not just interfaces.
But yes this is a stepping stone towards the eventual removal of the originals